sql/layers/layers.go

```diff
@@ -342,6 +342,50 @@ func IterateLayersWithBlockOps(
 			stmt.ColumnBytes(4, l.StateHash[:])
 			stmt.ColumnBytes(5, l.AggregatedHash[:])
 
+			inner := types.InnerBlock{}
+			_, err := codec.DecodeFrom(stmt.ColumnReader(7), &inner)
+			if err != nil {
+				return fn(l)
+			}
+
+			l.Block, err = types.NewExistingBlock(l.AppliedBlock, inner), nil
+			if err != nil {
+				derr = err
+				return false
+			}
+
+			return fn(l)
+		})
+	if err != nil {
+		return err
+	}
+	return derr
+}
+
+func Get(
+	db sql.Executor,
+	lid types.LayerID,
+	fn func(layer *Layer) bool,
+) error {
```

Comment:
If this is supposed to always return 1 layer, why not just return it?
```suggestion
func Get(
	db sql.Executor,
	lid types.LayerID,
) (layer *Layer, error) {
```